### PR TITLE
Feature: shrink docker - FS dependencies

### PR DESCRIPTION
**Pull request by m-reuter**
This PR removes several dependencies from the Docker images with FreeSurfer, shaving away around 2GB of data. 

It has not yet been fully tested for all parameter settings, but worked for default. 

This is addressing #138

### CODE REVIEW

**Comment by af-a** [comment]
I can't seem to build the recon_surf docker image (from Dockerfile_reconsurf) in its current state; I get the error:
```
Step 8/12 : RUN python$PYTHON_VERSION -m pip install -U git+https://github.com/Deep-MI/LaPy.git#egg=lapy &&     python$PYTHON_VERSION -m pip install simpleitk==2.1.1
 ---> Running in 7a64b0e1724e
Collecting lapy
  Cloning https://github.com/Deep-MI/LaPy.git to /tmp/pip-install-way_kyrs/lapy_37aa311933e94d5c914410d5e490bcc1
  Running command git clone -q https://github.com/Deep-MI/LaPy.git /tmp/pip-install-way_kyrs/lapy_37aa311933e94d5c914410d5e490bcc1
  fatal: unable to access 'https://github.com/Deep-MI/LaPy.git/': server certificate verification failed. CAfile: none CRLfile: none
WARNING: Discarding git+https://github.com/Deep-MI/LaPy.git#egg=lapy. Command errored out with exit status 128: git clone -q https://github.com/Deep-MI/LaPy.git /tmp/pip-install-way_kyrs/lapy_37aa311933e94d5c914410d5e490bcc1 Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement lapy (unavailable) (from versions: none)
ERROR: No matching distribution found for lapy (unavailable)
The command '/bin/sh -c python$PYTHON_VERSION -m pip install -U git+https://github.com/Deep-MI/LaPy.git#egg=lapy &&     python$PYTHON_VERSION -m pip install simpleitk==2.1.1' returned a non-zero code: 1
```
I think the issue is that `LaPy` is not found: `fatal: unable to access 'https://github.com/Deep-MI/LaPy.git/': server certificate verification failed.`

Tested this on two machines, with the same results:
1. Ubuntu 20.04.4 LTS (`Linux ... 5.13.0-30-generic #33~20.04.1-Ubuntu SMP Mon Feb 7 14:25:10 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`)
2. CentOS Linux 7 (Core) (`Linux ... 3.10.0-1160.49.1.el7.x86_64 #1 SMP Tue Nov 30 15:51:32 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`)

I was able to fix this by including the `ca-certificates` package, which was recently removed from the Dockerfile.
I think we may need to add that back in.

File: Docker/Dockerfile [hunk 1]
```diff
@@ -7,45 +7,46 @@ ENV DEBIAN_FRONTEND=noninteractive
 
 # Install custom libraries + freesurfer 6.0 (and necessary dependencies)
 RUN apt-get update && apt-get install -y --no-install-recommends \
-         build-essential \
-         cmake \
-         git \
-         vim \
-         wget \
-         ca-certificates \
```
Comment (af-a) [inline Docker/Dockerfile:15]: We may need to include `ca-certificates` again, due to the aforementioned build error.

File: Docker/Dockerfile_CPU [hunk 2]
```diff
@@ -7,61 +7,48 @@ ENV DEBIAN_FRONTEND=noninteractive
 
 # Install custom libraries + freesurfer 6.0 (and necessary dependencies)
 RUN apt-get update && apt-get install -y --no-install-recommends \
-         build-essential \
-         cmake \
-         git \
-         vim \
-         wget \
-         ca-certificates \
```
Comment (af-a) [inline Docker/Dockerfile_CPU:15]: Same

File: Docker/Dockerfile_reconsurf [hunk 3]
```diff
@@ -6,67 +6,54 @@ ARG PYTHON_VERSION=3.8
 ENV DEBIAN_FRONTEND=noninteractive
 
 # Install custom libraries + freesurfer 6.0 (and necessary dependencies)
-RUN apt-get update && apt-get install -y --no-install-recommends \
-         build-essential \
-         cmake \
-         git \
-         vim \
-         wget \
-         ca-certificates \
```
Comment (af-a) [inline Docker/Dockerfile_reconsurf:15]: Same

**Comment by af-a** [comment]
`recon-surf.sh` also seems to fail at the following recon-all call:
https://github.com/Deep-MI/FastSurfer/blob/c30362d5bfb219f93ccc706e5bafeb0bb221c556/recon_surf/recon-surf.sh#L528

I observe the error:
```
#--------------------------------------------                                   
#@# Fill Wed Apr  6 09:10:42 UTC 2022                                           
FREESURFER: Undefined variable.                                                 
Command exited with non-zero status 1                                           
@#@FSTIME  2022:04:06:09:06:13 recon-all N 7 e 268.75 S 3.49 U 264.61 P 99% M 1086920 F 0 R 1751580 W 0 c 493 w 931 I 0 O 10896 L 3.44 1.67 0.68
@#@FSLOADPOST 2022:04:06:09:10:42 recon-all N 7 3.14 2.83 1.4
```

This occurs in different flag combinations (--fsaparc --fstess, --fsaparc, --fstess, and the defaults).

**Comment by LeHenschel** [comment]
Yes, this is an error in FS7.2. At one point instead of FREESURFER_HOME they set the variable to FREESURFER inside the recon-all fill command (https://github.com/freesurfer/freesurfer/blob/fs-7.2/scripts/recon-all#L3028). We can either call mri_fill directly (`mri_fill -a ../scripts/ponscc.cut.log -segmentation aseg.presurf.mgz -ctab /opt/freesurfer/SubCorticalMassLUT.txt -auto-man filled.auto.mgz filled.mgz ../tmp/filled.edits.txt wm.mgz filled.mgz` ) in which case the error does not occur or just export FREESURFER=FREESURFER_HOME (set this in inside docker or during the call).

It is also a bit odd, that we get to the auto-man part. FS_ALLOW_FILLED_EDIT is turned on by default for some reason (https://github.com/freesurfer/freesurfer/blob/fs-7.2/scripts/recon-all#L454). This variable did not exist in FS6.0. Here, the mri_fill command was: `mri_fill -a ../scripts/ponscc.cut.log -xform transforms/talairach.lta -segmentation aseg.auto_noCCseg.mgz wm.mgz filled.mgz`. The filled.auto.mgz exists, but not sure where it is generated. This should only be for manual edits to filled.mgz

**Comment by m-reuter** [comment]
I'll update the PR with the ca-certificates. 

We should export the FREESURFER variable in recon-surf so that it is automatically done independent of the container or direct install. Can one of you do that? Also is this a know bug in FreeSurfer or do we need to report it?

I also noticed in some log output that label2vol complained that no --seed is given (probably it is called from inside another script and not directly from recon-all). This is probably also a bug in FS7.2. Take a look at the recon-surf log to see if you find it and then we need to check if we have to report (or fix) it in FreeSurfer.

**Comment by m-reuter** [comment]
I'll also add the FREESURFER export to this pull request.

**Comment by af-a** [comment]
> I also noticed in some log output that label2vol complained that no --seed is given (probably it is called from inside another script and not directly from recon-all). This is probably also a bug in FS7.2. Take a look at the recon-surf log to see if you find it and then we need to check if we have to report (or fix) it in FreeSurfer.

I do consistently see this message in the `recon-surf.log`s:
```
mri_label2vol supposed to be reproducible but seed not set
```

**Comment by LeHenschel** [comment]
The FREESURFER bug has been fixed in the dev version of FreeSurfer (so it should be known).

**Comment by m-reuter** [comment]
> I do consistently see this message in the `recon-surf.log`s:
> 
> ```
> mri_label2vol supposed to be reproducible but seed not set
> ```

Yes, that is what I mean. But 
- where exactly is this called from (maybe multiple places and I think not from recon-all but somewhere deeper)
- does that script allow passing a --seed and if so, why is it not passing the seed deeper to label2vol
- is this already a known bug and fixed in freesurfer dev or not

I would not worry about it too much now, but if it is unknown to them we should at least report it.

**Pull request merged**